### DISCUSSION TITLE
Settings via companion app

### DISCUSSION THREAD

**Discussion by jmlich** [InfiniTimeOrg/InfiniTime#1889] [Ideas]
It seems other smart watches moves implementation of some settings into companion app

see
https://github.com/piggz/harbour-amazfish/blob/7a3b0d20b0a8646599bd17feb96a163bd6f86a4b/ui/qml/pages/Settings-device.qml#L119
and
https://github.com/piggz/harbour-amazfish/blob/7a3b0d20b0a8646599bd17feb96a163bd6f86a4b/daemon/src/devices/huamidevice.cpp#L236

This might be usable for
- Weather - temperature unit (Celsius/Farnheit)
- Language
- Profile (Laud, Silent, Sleep)
- Alarm or multiple alarms
- Distance Unit
- Steps Goal
- Hear rate monitoring interval
- enabling/disabling debug logs
- date/time format

**Comment by SteveAmor** [comment]
I believe, for some, advanced features, moving into a companion app would be good from a UI/UX point of view and to save memory.  But the watch should have good functionality without any companion app - eg 12/24 hr format should remain in the watch, minimum one alarm etc.

**Comment by SteveAmor** [comment]
An alternative would be a text editable (edit offline) resource file (advancesettings.ini) that allows you to unlock hidden settings rather than taking up memory building extended UIs.

I suppose the question is, why would we move some of the settings to a companion app or "advanced settings" file?

**Reply by jmlich** [reply]
As you mentioned earlier. Reasons are better UX and saving some cpu/memory. It is question if is necessary to be able configure every single option in UI. There are some options which can be hidden from user or are difficult to implement within PineTime UI it self. For example customization of menu items order.

**Reply by jmlich** [reply]
Also the PineTimeStyle watch face settings is quite hidden. It took me long time to find out that it is there. The its whole settings is very uncomfortable.
